qutebrowser/mainwindow/tabwidget.py

```diff
@@ -570,10 +569,10 @@ def _text_to_width(text):
 
     def _tab_pinned(self, index: int) -> bool:
         """Return True if tab is pinned."""
-        try:
-            return self.tab_data(index, 'pinned')
-        except KeyError:
-            return False
+        if not 0 <= index < self.count():
+            raise IndexError("Tab index ({}) out of range ({})!".format(
+                index, self.count()))
+        return self.parent().widget(index).data.pinned
```

Comment:
Is it safe to rely on `parent` being the tabwidget? It's already being used in `current_tab` so I don't think this will change much.